opensearch-project/OpenSearch-Dashboards
# [ignoreFilterIfFieldNotInIndex] Parse query string filters to determine if fields match an index

### Description

For "query_string" filters, parse the query string as lucene to determine if the query's fields matches any given index. This is pertinent when the `ignoreFilterIfFieldNotInIndex` advanced setting is enabled, as filters are only applied to applicable indexes.

Also, for filter bar items, the "Warning" when a filter does not match any indexes has been updated to use the same logic.

### Issues Resolved

closes #6036 

## Screenshot



## Changelog
- feat: Parse query string filters to determine if fields match an index when `ignoreFilterIfFieldNotInIndex` is enabled

## Testing the changes

1. Enable `ignoreFilterIfFieldNotInIndex`
2. Go to a dashboard
3. Click Add Filter
4. Click Edit as Query DSL
5. Enter a "query string" filter, such as: `{ "query": { "query_string": { "query": "foo: bar" } } }`
6. Click Save
7. Notice that the filter should now be applied to matching indexes, whereas before it was not
8. Also notice that if the filter does not have an index pattern specifically associated with it, "Warning" will not appear in the filter item's label, whereas before it would

### Check List
